### DISCUSSION TITLE
Is it possible to animate an Icon on click ?

### DISCUSSION THREAD

**Discussion by abouroubi** [visgl/deck.gl#8071] [Q&A] [answered]
I want to recreate the Apple Maps effect when the user selects an icon on the map, a small animation is played and the icon gets bigger.

Is it possible do it with deck.gl ?

https://github.com/visgl/deck.gl/assets/11291350/0468b122-0a28-40b3-b11a-3924585fe359

**Comment by Pessimistress** [comment]
You can use [updateTriggers](https://deck.gl/docs/api-reference/core/layer#updatetriggers) to update the icon size and [transitions](https://deck.gl/docs/api-reference/core/layer#transitions) for smooth animation.

**Comment by abouroubi** [comment]
Thanks for the quick answer.
Are there some sample code on how to use those functions ? I'm a little bit lost with only the docs.

**Reply by Pessimistress** [reply, marked as answer]
https://codepen.io/Pessimistress/pen/ZEVbMYE?editors=001
